source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -312,6 +312,12 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const Envoy::Ssl::ContextConfig& c
           absl::StrCat("Failed to load certificate chain from ", ctx.cert_chain_file_path_));
     }
 
+    // The must staple extension means the certificate promises to carry
+    // with it an OCSP staple. https://tools.ietf.org/html/rfc7633#section-6
+    std::string staple_request_oid = "1.3.6.1.5.5.7.1.24";
+    auto must_staple = Utility::getX509ExtensionValue(*ctx.cert_chain_, staple_request_oid);
+    ctx.is_must_staple_ = must_staple.has_value();
```

Comment:
Should there also be a check to make sure the value contains the expected id?

Reply:
Yes, it looks like there aren't other options at the moment but to be correct we should check this for the must-staple octet value.